### PR TITLE
WD-11666 - refactor: remove nested components in ConfigPanel

### PR DESCRIPTION
**Pull request by vladimir-cucu**
## Done

- Removed nested components in `panels/ConfigPanel`.

## Details

- https://warthogs.atlassian.net/browse/WD-11666

### CODE REVIEW

File: src/panels/ConfigPanel/types.ts [hunk 1]
```diff
@@ -44,3 +46,22 @@ export enum Label {
 export enum TestId {
   PANEL = "config-panel",
 }
+
+export enum InlineErrors {
+  FORM = "form",
+  GET_CONFIG = "get-config",
+  SUBMIT_TO_JUJU = "submit-to-juju",
```
Comment (huwshimi) [inline src/panels/ConfigPanel/types.ts:53]: `FORM` and `SUBMIT_TO_JUJU` are only used inside ConfirmationDialog so they should be moved to a types file for that component.

It looks like a lot of of the values in the `Label` can also be moved to the new component.

**Review by huwshimi — COMMENTED** [review]
This adds new tests for the confirmation dialog component, but does not remove/modify any from ConfigPanel even though the logic has moved into the new component.

I think it would be worth taking a look at what is occurring in each component and making sure the tests are relevant to the logic in that "unit".

For example: in the config panel we might want to make sure that a particular confirmation is displayed under certain conditions, and in the confirmation dialogue we'd check the API calls that it makes when confirmed.

File: src/panels/ConfigPanel/types.ts (hunk 1 above)
Comment (vladimir-cucu) [reply src/panels/ConfigPanel/types.ts]: Nice catch! Everything that was no longer used in `ConfigPanel` was moved to `ConfirmationDialog`. Thanks for the review.

File: src/panels/ConfigPanel/ConfirmationDialog/ConfirmationDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    const response = await setApplicationConfig(appName, config);
+    const errors = response?.results?.reduce<string[]>((collection, result) => {
+      if (result.error) {
+        collection.push(result.error.message);
+      }
+      return collection;
+    }, []);
+    setSavingConfig(false);
+    setEnableSave(false);
+    setConfirmType(null);
```
Comment (vladimir-cucu) [inline src/panels/ConfigPanel/ConfirmationDialog/ConfirmationDialog.tsx:67]: This call might be redundant, as we already set the config type to `null` on line 106.

File: src/panels/ConfigPanel/ConfirmationDialog/ConfirmationDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+              // It is not possible to get to this point if these
+              // variables aren't set.
+              return;
+            }
+            void (async () => {
+              try {
+                for (const secretURI of requiredGrants) {
+                  await grantSecret(secretURI, [appName]);
+                }
+                setConfirmType(null);
```
Comment (vladimir-cucu) [inline src/panels/ConfigPanel/ConfirmationDialog/ConfirmationDialog.tsx:150]: This call might be redundant as well, as we set the confirm type to `null` on line 137.

**Review by huwshimi — APPROVED** [review]
Just a couple of small changes

**Review by huwshimi — APPROVED** [review]
That fix looks good.

**Pull request merged**